fideslib/core/config.py

```diff
@@ -49,22 +49,24 @@ class DatabaseSettings(FidesSettings):
     port: str = "5432"
     test_db: str = "test"
 
-    sqlalchemy_database_uri: Optional[PostgresDsn] = None
-    sqlalchemy_test_database_uri: Optional[PostgresDsn] = None
+    sqlalchemy_database_uri: Optional[str] = None
+    sqlalchemy_test_database_uri: Optional[str] = None
 
     @validator("sqlalchemy_database_uri", pre=True)
     @classmethod
     def assemble_db_connection(cls, v: Optional[str], values: Dict[str, str]) -> str:
         """Join DB connection credentials into a connection string"""
         if isinstance(v, str):
             return v
-        return PostgresDsn.build(
-            scheme="postgresql",
-            user=values["user"],
-            password=values["password"],
-            host=values["server"],
-            port=values.get("port"),
-            path=f"/{values.get('db') or ''}",
+        return str(
+            PostgresDsn.build(
+                scheme="postgresql",
+                user=values["user"],
+                password=values["password"],
+                host=values["server"],
+                port=values.get("port"),
+                path=f"/{values.get('db') or ''}",
+            )
```

Comment:
For both of these validators should we do the `PostgresDsn.build(...)` as it's own step wrapped in a try/except, and then convert the result to a string after successfully building the dsn? The reason I say this is if there is an error currently it will look something like: `TypeError: AnyUrl.build() missing 1 required keyword-only argument: 'host'`, which makes sense in the context of a PostgresDsn, but with the user expecting a string this error is confusing. With a try/catch we could return some error that makes sense in the context of a string.

Reply:
@sanders41 I generally understand what you're saying but not 100%, can you pseudo-code it or commit it directly?

Happy to make the change for better user-experience, just want to totally understand it first

Reply:
This was harder than I expected to come up with a user friendly message that is still useful in tracking down the error. I'm going to approve it as is and we can revisit later if we get questions about the error message.